CHANGELOG.md

```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Fixed
 - Previously, every peer list reported itself as a "single" peer list for
   purposes of debugging, instead of its own name.
+- Metrics emit `CodeResourceExhausted` as a client error and `CodeUnimplemented`
```

Comment:
Resource exhausted I’ll buy. That’s clearly a bug.

Unimplemented I’m not so sure about. This should in practice only occur when a client calls a method that does not exist, and while it seems possible that it’s the server’s fault, it seems more likely to me to be the client’s. What do you think?

Reply:
I can see both sides to this. The server probably shouldn't provide an RPC interface if it's not yet implemented. Though the client should probably understand which endpoints are callable. 

I think this change (unimplemented as server error) would be the least surprising to a user, so I'd lean to that side. The server ultimately determines what is implemented or not, and the client's only reaction can be to give up or try another server.